### PR TITLE
WT-7292 update webhook subscribe callback definition

### PR DESCRIPTION
**Pull request by ShansOwn**
Changed errors codes and added descriptions to content for errors.

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -307,10 +307,22 @@ paths:
...
-                '420':
-                  description: Returns this code to indicate a transient error. The web hook post request will be retried after a delay.
-                '428':
-                  description: Returns this code when permanent failure has occurred, to signify that no retries should be made.
+                '400':
+                  description: >
+                    Indicates that permanent failure has occurred, to signify that no retries should be made.
+                    Returns a user-friendly error message which describes the problem.
+                  content:
+                    text/plaint:
```
Comment (vyshane) [inline spec/openapi.yaml]: Do you mean `text/plain`?

File: spec/openapi.yaml [hunk 2]
```diff
@@ -307,10 +307,22 @@ paths:
...
+                      schema:
+                        type: string
+                '500':
+                  description: >
+                    Indicates a transient error. The web hook post request will be retried after a delay.
+                    Returns a user-friendly error message which describes the problem.
+                  content:
+                    text/plaint:
```
Comment (vyshane) [inline spec/openapi.yaml]: Do you mean `text/plain`?

**Review by vyshane — CHANGES_REQUESTED** [review]
Couple of typos.

File: spec/openapi.yaml (hunk 1 above)
Comment (ShansOwn) [reply spec/openapi.yaml]: Sure. Fixed.

File: spec/openapi.yaml (hunk 2 above)
Comment (ShansOwn) [reply spec/openapi.yaml]: Sure. Fixed.

**Comment by vyshane** [comment]
There are serious implications to changing the HTTP codes returned by connector webhooks. This means that we will need to update the connector library, create new connector builds and update every single connector currently running in production. I vote for leaving them unchanged.

**Review by vyshane — CHANGES_REQUESTED** [review]
Do we really want to update webhook response codes?

**Comment by ShansOwn** [comment]
> Do we really want to update webhook response codes?

They've been already updated [here](https://github.com/wisetime-io/wisetime-connector-java/commit/40454a02bf8de8e4d7c21e38ef0b02e0276140c0#diff-839fd0611f7f69f4ad141ddca0bc90b4R96). I've just updated the spec according to them

**Comment by vyshane** [comment]
I see. We need to think this one through. @pfilippi24 will the Connect API backend keep support for old webhooks? We can't break backwards compatibility on this one without a great deal of coordination and pain.

**Review by vyshane — APPROVED** [review]

**Review by pfilippi24 — APPROVED** [review]

**Pull request merged**